modules/global-config/src/main/java/org/mule/runtime/globalconfig/internal/cluster/ClusterConfig.java

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.globalconfig.internal.cluster;
+
+import static java.lang.Integer.MAX_VALUE;
+import static java.lang.Integer.valueOf;
+import static java.lang.String.format;
+import static java.util.Optional.empty;
+import static java.util.Optional.of;
+import org.mule.api.annotation.NoImplement;
+import org.mule.maven.client.api.model.Authentication;
+import org.mule.maven.client.api.model.MavenConfiguration;
+import org.mule.maven.client.api.model.RemoteRepository;
+import org.mule.maven.client.api.model.RepositoryPolicy;
+import org.mule.runtime.api.exception.MuleRuntimeException;
+import org.mule.runtime.api.i18n.I18nMessageFactory;
+import org.mule.runtime.container.api.MuleFoldersUtil;
+import org.mule.runtime.globalconfig.api.EnableableConfig;
+import org.mule.runtime.globalconfig.api.exception.RuntimeGlobalConfigException;
+
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigObject;
+
+import java.io.File;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Comparator;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Configuration builder for {@link MavenConfiguration} instances.
+ */
+public class ClusterConfig {
+
+  static final String POSITION = "position";
+  EnableableConfig objectStoreConfig;
+  EnableableConfig lockFactoryConfig;
+  EnableableConfig timeSupplierConfig;
+
+  public EnableableConfig getObjectStoreConfig() {
+    return objectStoreConfig;
+  }
+
+  public EnableableConfig getLockFactoryConfig() {
+    return lockFactoryConfig;
+  }
+
+  public EnableableConfig getTimeSupplierConfig() {
+    return timeSupplierConfig;
+  }
+
+  /**
+   * @param config the maven configuration set by the user
+   * @return a {@link MavenConfiguration} created by using the user configuration and default values set by mule.
+   */
+  public static MavenConfiguration buildMavenConfig(Config config) {
```

Comment:
was all this logic moved from somepace else? this is maven stuff but the pr title talks about cluster config